### ISSUE TITLE
linux paa issues

### ISSUE THREAD

**Issue by dener3id** [PCSX2/pcsx2#3666]
sudo add-apt-repository ppa:gregory-hainaut/pcsx2.official.ppa  doesn't support focal. also as far as I know 1.4 is the last version that works with ds3 correctly on Linux it should be added to the paa for testing and trouble shooting otherwise I have no way to confirm issues can't compile on Linux, there's something seriously wrong with the Linux compile on Linux also at least one or more depends isn't the repos for my OS. 

I'd recommend providing a .deb, and one containing all depends as different ubuntu based systems can all be slightly different. if nothing else a stand alone .deb download. other Linux systems have ways to translate .debs.

**Comment by RedDevilus** [comment]
For now it's best that you compile 1.6 or the recent 1.7 devbuilds till someone provides a stable 1.6 PPA.

**Comment by dener3id** [comment]
> For now it's best that you compile 1.6 or the recent 1.7 devbuilds till someone provides a stable 1.6 PPA.

witch of the two is better? and how I deal with the depends?

**Comment by Mrlinkwii** [comment]
theirs a PPA for focal at pcsx2-team/pcsx2-daily which is the latest Dev build  for focal if you dont want to complie it

**Comment by RedDevilus** [comment]
Stability and extensively tested = 1.6 stable
Latest stuff but can have problems = 1.7 dev build

**Comment by dener3id** [comment]
how to I deal with the depends?

**Comment by Mrlinkwii** [comment]
for compiling pcsx2 on linux use this guide https://github.com/PCSX2/pcsx2/wiki/Installing-on-Linux

**Comment by RedDevilus** [comment]
Should help you
https://github.com/PCSX2/pcsx2/wiki/Installing-on-Linux

This is not a support location, use the forums next time.

**Comment by dener3id** [comment]
> Should help you
> https://github.com/PCSX2/pcsx2/wiki/Installing-on-Linux
> 
> This is not a support location, use the forums next time.

dosn't show up

**Comment by beaumanvienna** [comment]
On focal you need to compile against GTK3, which is accomplished by using `-DGTK3_API=TRUE`. I would try with these dependencies:
`cmake:i386 debhelper:i386 dpkg-dev:i386 libaio-dev:i386 libasound2-dev:i386 libgl1-mesa-dev:i386 libglu1-mesa-dev:i386 liblzma-dev:i386 libpcap0.8-dev:i386 
libpng-dev:i386 libpulse-dev:i386 :i386 libsdl2-dev:i386 libsoundtouch-dev:i386 libwxbase3.0-dev:i386 libwxgtk3.0-gtk3-dev:i386 libgtk-3-dev:i386 libxml2-dev:i386 libx11-dev:i386 
locales:i386 portaudio19-dev:i386 zlib1g-dev:i386 libx11-xcb-dev:i386 libbz2-dev:i386 libudev-dev:i386 `

On the wiki page @RedDevilus had posted, all the way at the bottom the (essential) compile instructions are 
```
cmake -DCMAKE_TOOLCHAIN_FILE=cmake/linux-compiler-i386-multilib.cmake \
-DCMAKE_BUILD_TYPE=Release -DGTK3_API=TRUE .. 
```

In the [issue ticket for compiling the x64 version](https://github.com/PCSX2/pcsx2/issues/3646#issuecomment-685875595), @Shoegzer had explained the important options very well.

Just on a side note, I would recommend to go with the x64 version. In this version only the recompilers are slightly different. The changes have been tested quite a lot by now and the port seems to be stable. Especially under Ubuntu, dealing with multiple versions of the same lib, be it GTK2 and GTK3, or a normal and a i386 version can be tricky. I test under both Ubuntu and Arch, and Arch can deal much better with those parallel installations. When I work for PCSX2 on Ubuntu 18.04, I compile in a virtual machine running Lubuntu. It might be ok to install user libs for x64 and i386 in parallel  under Ubuntu, but I could not get the developer libs installed in parallel and make it work. Arch can handle this much better. So long story short, unless you have a specific reason to use the 32-bit version, I would recommend to go with the new version.

Good luck!

**Comment by beaumanvienna** [comment]
@dener3id Do you need any more help on this? We did not hear back from you in two weeks.

 If you're stuck, you could try building PCSX2 in a chroot jail. As mentioned above, I also noticed under Ubuntu that compiling PCSX2 does not work when the developer libs needed to build it are installed for both i386 and x64 at the same time. I used a virtual machine to compile the 32-bit version. A chroot environment might be easier to setup and compile faster. 

Please let us know if you have any feedback or if this item can be closed.

**Comment by beaumanvienna** [comment]
@MrCK1 I created a wiki about compiling in a clean chroot in my project. Maybe merge it into the PCSX2 wiki?
https://github.com/beaumanvienna/marley#having-trouble-compiling

**Comment by OdinVex** [comment]
Many distros have begun to remove x86 (32-bit). The move stems from an industry shift (from Microsoft with Windows to Linux and Mac) to slowly begin phasing out *new* x86 software development for x64 (64-bit) while updating past software to x64. I've run into nothing but trouble attempting to compile PCSX2 on my distro because of package conflicts because the host is x64, not to mention the trouble with trying to co-exist with oibaf-graphics-drivers PPA. If PCSX2 isn't ready for x64-only could someone write up a small doc about what development would be needed to begin moving to x86+x64 or x64-only compilation?

**Comment by tellowkrinkle** [comment]
While we use 32-bit for official builds, you're free to compile PCSX2 for x64 for yourself.  Everything should be working on Linux with the exception of mipmapping in the SW renderer and support for pre-AVX CPUs

**Comment by OdinVex** [comment]
> While we use 32-bit for official builds, you're free to compile PCSX2 for x64 for yourself. Everything should be working on Linux with the exception of mipmapping in the SW renderer and support for pre-AVX CPUs

Perhaps I have a not clarified the issue. x86 (32-bit) is being phased out for Linux. Many distros have not only disabled it by default but have actually stopped providing x86 builds for libraries. It has become rather impossible for some distros and rather difficult for some others to compile, let alone run...

**Comment by refractionpcsx2** [comment]
> > While we use 32-bit for official builds, you're free to compile PCSX2 for x64 for yourself. Everything should be working on Linux with the exception of mipmapping in the SW renderer and support for pre-AVX CPUs
> 
> Perhaps I have a not clarified the issue. x86 (32-bit) is being phased out for Linux. Many distros have not only disabled it by default but have actually stopped providing x86 builds for libraries. It has become rather impossible for some distros and rather difficult for some others to compile, let alone run...

Then do what tellow said and build 64bit.

**Comment by OdinVex** [comment]
> Then do what tellow said and build 64bit.
There doesn't seem to be any documentation on compiling for x64. For most projects, simply changing targets works, but this one seems geared around x86 with a lot of explicit 'x86-ness' patched here and there for compilation (it isn't as unified, is all). Perhaps it would be best for someone more familiar with the project to list anything out of the ordinary about compiling for x64?

**Comment by F0bes** [comment]
> > Then do what tellow said and build 64bit.
> There doesn't seem to be any documentation on compiling for x64. For most projects, simply changing targets works, but this one seems geared around x86 with a lot of explicit 'x86-ness' patched here and there for compilation (it isn't as unified, is all). Perhaps it would be best for someone more familiar with the project to list anything out of the ordinary about compiling for x64?

Look in build.sh, the argument is 
--no-cross-multilib 

Or just use cmake with -DCMAKE_BUILD_TYPE=Release

**Comment by OdinVex** [comment]
> --no-cross-multilib

--no-cross-multilib results in a Ninja error while --no-cross-multilib --release compiled fine but has no built binaries. At least a starting point.

**Comment by F0bes** [comment]
> > --no-cross-multilib
> 
> --no-cross-multilib results in a Ninja error while --no-cross-multilib --release compiled fine but has no built binaries. At least a starting point.

Okay glad it kinda helped. I think that this should be taken either to the discord or the forums. It's now off topic from the original issue.

https://discord.com/invite/TCz3t9k

**Comment by OdinVex** [comment]
> > > --no-cross-multilib
> > 
> > 
> > --no-cross-multilib results in a Ninja error while --no-cross-multilib --release compiled fine but has no built binaries. At least a starting point.
> 
> Okay glad it kinda helped. I think that this should be taken either to the discord or the forums. It's now off topic from the original issue.
> 
> https://discord.com/invite/TCz3t9k

I'll wade through for now, I don't Discord, but thank you. I still think I should edit any last post of mine after success with a short explanation for others who may be seeking to compile because of PPA age/issues.

**Comment by tellowkrinkle** [comment]
> --no-cross-multilib results in a Ninja error (hates /usr/bin/cc despite trying with gcc/g++ 9 and 10 with update-alternatives) while --no-cross-multilib --release compiled fine but has no built binaries. At least a starting point.

Our build script will disable parts if you don't have the required packages installed.  This can end up disabling literally everything.  Compile with cmake directly (instead of build.sh) and look through the cmake output for messages saying it's not building things due to dependencies missing.  You'll need GTK3 (not 2), I'm not sure if the documentation has been updated to reflect that

**Comment by OdinVex** [comment]
> Our build script will disable parts if you don't have the required packages installed. This can end up disabling literally everything. Compile with cmake directly (instead of build.sh) and look through the cmake output for messages saying it's not building things due to dependencies missing. You'll need GTK3 (not 2), I'm not sure if the documentation has been updated to reflect that

I noticed that and installed a few (x64, I don't multiarch) packages. Compilation was still unsuccessful but only because PCSX2 keeps targetting x86 instead of x64 (target).

**Comment by lightningterror** [comment]
Closing as we have appimages 32/64.

**Comment by OdinVex** [comment]
^
Hurray for more bloat 'designed to solve DLL hell'...*sigh* I'll just keep compiling on my own.

**Comment by kenshen112** [comment]
@OdinVex The ppa is still maintained as actively as ever by the pcsx2 team. Though it's not the above link it's at https://launchpad.net/~pcsx2-team/+archive/ubuntu/pcsx2-daily

**Comment by OdinVex** [comment]
@kenshen112 It wasn't, until recently. I got into the habit of building it myself for x64 in 2020 because of the lack of x64 builds with developers refusing to go x64 despite...well, just look around as a dev, eh. :)

**Comment by RedDevilus** [comment]
Where did you see outright refusal for x64 builds? I can believe lower priority and over what time period are you even talking about?

**Comment by OdinVex** [comment]
> Where did you see outright refusal for x64 builds? I can believe lower priority and over what time period are you even talking about?

Forums, I posted about two years ago, got flat-out told x64 builds were not happening. Since then, I wrote a small shell script to pull the git, compile, install, while maintaining last-known-good compilations.
